builtin/providers/aws/resource_aws_elasticsearch_domain.go

```diff
@@ -274,6 +275,22 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 
 	d.Set("arn", *ds.ARN)
 
+	// get tags
+	type ListTagsRequest struct {
+	    ARN string
+	}
```

Comment:
The AWS SDK already defines this for you, so you don't have to make your own struct, you can just use theirs:

```go
req := &elasticsearch.ListTagsInput{
	ARN:   aws.String(ds.ARN),
}
```